lib/baremetalbasetest.pm

```diff
@@ -0,0 +1,269 @@
+        if ($response->{status} == 200) {
+            set_var('BAREMETAL_LOCK_TOKEN', $response->{content});
+            break;
+        } elsif ($response->{status} == 412) {
+            if ($retries != 0) {
+                $retries--;
+                sleep(60);
+            } else {
+                record_info('lock timeout', 'failed to acquire the host lock. giving up!', 'fail');
+                die('timeout acquiring the host lock - host is not being freed in time');
+            }
+        } else {
+            record_info('lock failure', "failed to acquire the host lock. got HTTP status $response->{status} and message $response->{reason}", 'fail');
+            die("baremetalbasetest::host_lock(): failed to acquire the host lock. got HTTP status $response->{status} and message $response->{reason}");
+        }
+    }
+
+}
+
+=head2 host_is_locked
+
+ host_is_locked() 
+
+Check if the SUT is locked in the support service
+=cut
+sub host_is_locked {
+    my $host        = get_required_var('SUT_IP');
+    my $ip          = inet_ntoa(inet_aton($host));
+    my $http_server = get_required_var('IPXE_HTTPSERVER');
+    my $url_status  = "$http_server/v1/host_lock/lock_state/$ip";
+
+    my $response = HTTP::Tiny->new->request('GET', $url_status, {content => '', headers => {'content-type' => 'text/plain'}});
+    if ($response->{status} != 200) {
+        diag("baremetalbasetest::host_is_locked(): unable to get lock state for $host: $response->{status} ($response->{reason}");
+    } else {
+        return unless $response->{content} eq 'locked';
+    }
+}
+
+=head2 host_unlock
+
+ host_unlock() 
+
+Unlock the SUT from the baremetal support service. 
+=cut
+sub host_unlock {
+    my $host        = get_required_var('SUT_IP');
+    my $ip          = inet_ntoa(inet_aton($host));
+    my $http_server = get_required_var('IPXE_HTTPSERVER');
+    my $token       = get_var(BAREMETAL_LOCK_TOKEN);
+    my $url_status  = "$http_server/v1/host_lock/lock_state/$ip";
+    if ($token == '') {
+        diag("baremetalbasetest::host_unlock(): no token available, is this machine locked by me?");
+        return;
+    }
+    my $url      = "$http_server/v1/host_lock/lock/$ip/$token";
+    my $response = HTTP::Tiny->new->request('PUT', $url_status, {content => '', headers => {'content-type' => 'text/plain'}});
+    if ($response->{status} != 200) {
+        diag("baremetalbasetest::host_unlock(): unable to unlock $host: $response->{status} ($response->{reason}");
+    } else {
+        set_var('BAREMETAL_LOCK_TOKEN', '');
+    }
+}
+
+sub ipmitool {
+    my ($cmd) = @_;
+
+    my @cmd = ('ipmitool', '-I', 'lanplus', '-H', $bmwqemu::vars{IPMI_HOSTNAME}, '-U', $bmwqemu::vars{IPMI_USER}, '-P', $bmwqemu::vars{IPMI_PASSWORD});
+    push(@cmd, split(/ /, $cmd));
+
+    my ($stdin, $stdout, $stderr, $ret);
+    print @cmd;
+    $ret = IPC::Run::run(\@cmd, \$stdin, \$stdout, \$stderr);
+    chomp $stdout;
```

Comment:
I prefer `$stdout ~= s/\s+$//; ` or `$stdout = Mojo::Util:trim($stdout);` as it also removes white space characters which aren't equal to system newline (`$/`)